### PR TITLE
Progress Bar for Scan created on worker page. Fixes #373

### LINKED ISSUE #373: Show progress of scan
On the Workers page, show progress with the total number of plug-ins selected and the number of the plug-in that is currently running.

For example:  (12/50)
Running the 12th plug-in of the 50 that were selected.

### PR DESCRIPTION
**Pull request by saganshul**
@delta24  @7a @DePierre Please Review. Fixes #373
As Discussed, I created Progress Bar on target page. I also added which worker is working on which plugin as suggested by @delta24.  
PFA screenshots.
screenshot from 2016-03-18 03-19-48

### CODE REVIEW

**Comment by sachinkamath** [comment]
That's beautiful @saganshul . Works very well locally :+1: :dancer:

**Comment by delta24** [comment]
@saganshul: Right now, keep the progress bar only in the workers page. I need to think of a more elegant UI fix to keep it in all targets page atm.

**Comment by saganshul** [comment]
@delta24 , okay! I will shift it again to worker's page.

**Comment by saganshul** [comment]
@delta24 @7a  Please Review.

File: framework/interface/api_handlers.py [hunk 1]
```diff
@@ -535,6 +535,27 @@ def delete(self, target_id=None, plugin_group=None, plugin_type=None, plugin_cod
             cprint(e.parameter)
             raise tornado.web.HTTPError(400)
```
Comment (delta24) [inline framework/interface/api_handlers.py]: 2 blank lines between 2 classes

File: framework/db/poutput_manager.py [hunk 2]
```diff
@@ -26,6 +26,13 @@ def PluginOutputExists(self, plugin_key, target_id):
         count = self.db.session.query(models.PluginOutput).filter_by(target_id=target_id, plugin_key=plugin_key).count()
         return (count > 0)
 
+    def PluginCountOutput(self):
+        complete_count = self.db.session.query(models.PluginOutput).count()
+        left_count = self.db.session.query(models.Work).count()
+        dict_list = []
```
Comment (delta24) [inline framework/db/poutput_manager.py]: Why not a simple dictionary as return value?

File: framework/interface/urls.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ def get_handlers():
         tornado.web.url(r'/api/targets/([0-9]+)/transactions/zconsole/?$', api_handlers.ZestScriptHandler, name='zest_console_api_url'),
         tornado.web.url(r'/api/targets/([0-9]+)/transactions/([0-9]+)/zest/?$', api_handlers.ZestScriptHandler, name='zest_api_url'),
         tornado.web.url(r'/api/plugins/?' + plugin_group_re + '/?' + plugin_type_re + '/?' + plugin_code_re + '/?$', api_handlers.PluginDataHandler, name='plugins_api_url'),
+        tornado.web.url(r'/api/plugins' + '/progress' + '/?$', api_handlers.ProgressBarHandler, name='poutput_count'),
```
Comment (delta24) [inline framework/interface/urls.py]: Make this `tornado.web.url(r'/api/plugins/progress/?$', api_handlers.ProgressBarHandler, name='poutput_count')`

File: framework/db/poutput_manager.py (hunk 2 above)
Comment (saganshul) [inline framework/db/poutput_manager.py]: yes, you are right. I will correct it now.

File: framework/db/poutput_manager.py [hunk 4]
```diff
@@ -26,6 +26,12 @@ def PluginOutputExists(self, plugin_key, target_id):
         count = self.db.session.query(models.PluginOutput).filter_by(target_id=target_id, plugin_key=plugin_key).count()
         return (count > 0)
 
+    def PluginCountOutput(self):
+        complete_count = self.db.session.query(models.PluginOutput).count()
+        left_count = self.db.session.query(models.Work).count()
+        results = {'complete_count':complete_count, 'left_count':left_count}
```
Comment (delta24) [inline framework/db/poutput_manager.py]: Space after `:` and value in a dictionary - PEP8

**Pull request merged**